internal/compiler/compiler.go

```diff
@@ -875,6 +877,35 @@ func (c *compiler) expr(expr ast.Expr) {
 	}
 }
 
+// Generate a Concat opcode or, if possible compact multiple `Concat` into one `ConcatN`
+func (c *compiler) concatOp(e *ast.BinaryExpr) {
+	var parts []ast.Expr
+
+	for {
+		if be, ok := e.Left.(*ast.BinaryExpr); ok && be.Op == lexer.CONCAT {
+			parts = append(parts, e.Right)
+			e = be
+		} else {
+			parts = append(parts, e.Right)
+			parts = append(parts, e.Left)
+			break
+		}
+	}
+
+	if len(parts) == 2 {
+		c.expr(parts[1])
+		c.expr(parts[0])
+		c.add(Concat)
+		return
+	}
+
+	for i := len(parts) - 1; i >= 0; i-- {
```

Comment:
Might be good to add a comment here that `// we appended the values in reverse above`.